interface/modules/custom_modules/oe-module-ehi-exporter/src/EhiExporter.php

```diff
@@ -0,0 +1,723 @@
+        $zipOutput = $tempDir . DIRECTORY_SEPARATOR . $zipName;
+        $openStatus = $zip->open($zipOutput, \ZipArchive::CREATE | \ZipArchive::OVERWRITE);
+        if ($openStatus == false) {
+            throw new \RuntimeException("Failed to open zip archive at location " . $zipOutput);
+        }
+        foreach ($exportedResult->exportedTables as $result) {
+            if ($this->shouldExportAdditionalAssets($result->tableName)) {
+                $this->exportAdditionalAssets($zip, $result->tableName);
+            }
+            $taskResultContents = $this->getCsvFileContents($exportState, $result->tableName);
+            $addedToZip = $zip->addFromString($result->tableName . '.csv', $taskResultContents);
+            if (!$addedToZip) {
+                $this->logger->errorLogCaller("Failed to add " . $result->tableName . " to zip file");
+                throw new \Exception("Failed to add " . $result->tableName . " to zip file");
+            }
+        }
+        if ($jobTask->ehiExportJob->include_patient_documents) {
+            $this->addPatientDocuments($exportedResult, $zip, $jobTask->getPatientIds());
+        }
+        $this->addDocumentationReadme($zip);
+        $saved = $zip->close();
+        if (!$saved) {
+            $this->logger->errorLogCaller("Failed to save zip file ", ['zipName' => $zipName]);
+            throw new \Exception("Failed to generate zip file for job " . $jobTask->ehi_task_id . " zip status is " . $zip->status);
+        }
+        unset($zip);
+        $document = $this->createDatabaseDocumentFromZip($jobTask, $zipOutput, $zipName);
+        // now we remove the zip file
+        if (!unlink($zipOutput)) {
+            $this->logger->errorLogCaller("Failed to EHI zip file export", ['zipName' => $zipOutput]);
+        }
+        $this->clearResultFilesForJob($jobTask, $exportState);
+        return $document;
+    }
+
+    private function clearResultFilesForJob(EhiExportJobTask $jobTask, ExportState $state)
+    {
+        $tempDir = $state->getTempSysDir();
+        // grab list of files in the directory
+        // unlink each file
+        $files = glob($tempDir . '/*'); // get all file names
+        if ($files === false) {
+            $this->logger->errorLogCaller("Failed to retrieve file list from temporary directory", ['tempDir' => $tempDir]);
+            return;
+        }
+        foreach ($files as $file) { // iterate files
+            if (is_file($file)) {
+                unlink($file); // delete file
+            }
+        }
+    }
+
+    private function getCsvFileContents(ExportState $state, string $tableName)
+    {
+        // now we need to decrypt the contents and add them to the export.
+        $filePath = $state->getTempSysDir() . DIRECTORY_SEPARATOR . $tableName . '.csv';
+        if (file_exists($filePath)) {
+            $contents = file_get_contents($filePath);
+            return $this->cryptoGen->decryptStandard($contents);
```

Comment:
when doing encryptStandard or decryptStandard to item on the drive then you want to use the keys in the databsae, which will look like this:
```
return $this->cryptoGen->decryptStandard($contents, null, 'database');
```

Reply:
Should be resolved now.